samples/core/blur/blur.cpp

```diff
@@ -0,0 +1,745 @@
+                         0, output_image.pixels.data());
+
+    if (verbose) print_timings(end - start, passes);
+
+    // write output file
+    finalize_blur();
+}
+
+void BlurCppExample::load_device()
+{
+    // Create context
+    context = cl::sdk::get_context(dev_opts.triplet);
+
+    device = context.getInfo<CL_CONTEXT_DEVICES>().at(0);
+    queue = cl::CommandQueue(context, device, cl::QueueProperties::Profiling);
+    cl::CommandQueue::setDefault(queue);
+
+    cl::Platform platform{
+        device.getInfo<CL_DEVICE_PLATFORM>()
+    }; // https://github.com/KhronosGroup/OpenCL-CLHPP/issues/150
+
+    if (!diag_opts.quiet)
+    {
+        std::cout << "Selected platform: "
+                  << platform.getInfo<CL_PLATFORM_VENDOR>() << "\n"
+                  << "Selected device: " << device.getInfo<CL_DEVICE_NAME>()
+                  << "\n"
+                  << std::endl;
+    }
+}
+
+void BlurCppExample::read_input_image()
+{
+    /// If file not provided in command line, create a default one.
+    if (blur_opts.in.empty())
+    {
+        std::string fname("andrew_svk_7oJ4D_ewB7c_unsplash.png");
+
+        std::cout << "No file given, use standard image " << fname << std::endl;
+
+        const char* fcont = (const char*)andrew_svk_7oJ4D_ewB7c_unsplash_png;
+        const size_t fsize = andrew_svk_7oJ4D_ewB7c_unsplash_png_size;
+
+        std::fstream f(fname, std::ios::out | std::ios::binary);
+        if (!f.is_open())
+        {
+            throw std::runtime_error{ std::string{
+                "Cannot create a default image: open "
+                "andrew_svk_7oJ4D_ewB7c_unsplash_png" } };
+        }
+        f.write(fcont, fsize);
+        f.close();
+
+        blur_opts.in = fname;
+    }
+
+    input_image = cl::sdk::read_image(blur_opts.in.c_str(), nullptr);
+    image_size = { (cl::size_type)input_image.width,
+                   (cl::size_type)input_image.height };
+    width = input_image.width;
+    height = input_image.height;
+}
+
+void BlurCppExample::prepare_output_image()
+{
+    output_image.width = input_image.width;
+    output_image.height = input_image.height;
+    output_image.pixel_size = input_image.pixel_size;
+    output_image.pixels.clear();
+    output_image.pixels.reserve(sizeof(unsigned char) * output_image.width
+                                * output_image.height
+                                * output_image.pixel_size);
+}
+
+std::tuple<bool, bool, bool> BlurCppExample::query_capabilities()
+{
+    // 1) query image support
+    if (!device.getInfo<CL_DEVICE_IMAGE_SUPPORT>())
+    {
+        cl::util::detail::errHandler(CL_INVALID_DEVICE, nullptr,
+                                     "No image support on device!");
+    }
+
+    // 2) query if the image format is supported and change image if not
+    format = set_image_format();
+
+    // 3) query if device have local memory and its size
+    bool use_local_mem =
+        (device.getInfo<CL_DEVICE_LOCAL_MEM_TYPE>() == CL_LOCAL);
```

Comment:
Minor: this isn't querying the local memory size.  Note, it looks like the C version doesn't query the local memory size either.

I don't think we need to query the local memory size here, so maybe just fix the comment?